app/src/main/java/com/codesoom/assignment/application/AuthenticationService.java

```diff
@@ -0,0 +1,76 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserLoginData;
+import com.codesoom.assignment.errors.EmailNotFoundException;
+import com.codesoom.assignment.errors.UnauthorizedException;
+import com.codesoom.assignment.errors.WrongPasswordException;
+import com.codesoom.assignment.utils.JwtUtil;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.security.SignatureException;
+import org.springframework.stereotype.Service;
+
+/**
+ * 유저 인증 로직 담당.
+ */
+@Service
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private final UserRepository userRepository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository userRepository) {
+        this.jwtUtil = jwtUtil;
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * 토큰을 decode한 후 속성정보를 반환한다.
+     *
+     * @param accessToken 인증토큰
+     * @return 속성 정보
+     * @throws UnauthorizedException 인증에 실패한 경우
+     */
+    public Long parsetoken(String accessToken) {
+        if (accessToken == null || accessToken.isBlank()) {
+            throw new UnauthorizedException(accessToken);
+        }
+        try {
+            Claims claims = jwtUtil.decode(accessToken);
+            return claims.get("userId", Long.class);
+        } catch (SignatureException e) {
+            throw new UnauthorizedException(accessToken);
+        }
+    }
+
+    /**
+     * 로그인 데이터로 유저 정보 조회후, userID로 인증토큰을 반환한다.
```

Comment:
구현 코드를 읽어보면 `userId`를 사용해 `encode`하고 있지만, 그것은 메소드 안쪽의 일입니다. javadoc을 작성할 때에는 마치 메소드 안쪽을 알지 못하는 것처럼 작성해야 좋습니다.

```suggestion
     * 로그인 데이터로 유저 정보 조회후, 사용자 정보로 생성한 인증토큰을 반환한다.
```

Reply:
@johngrib  그 종립님 블로그 보고 왔는데, 그래야 하는 이유는, javadoc을 작성할시에는 그 메소드의 책임과 역할을 알게끔 해야하므로, 구체적인 구현 내용이 들어가면  안되기 때문인 것 같습니다.

궁금한 것은 이제 이러한 원칙들은 java를 만든 oracle에서 정한건가요? 아니면 개발자들 사이의 암묵적 합의 사항인가요?

Reply:
이건 오라클하고는 아무런 관련이 없어요. 개발자들끼리 합의한 것도 아닙니다. 이건 객체지향 원칙이 주석에도 적용된 것 뿐이에요.

Reply:
@johngrib 아하 그렇군요.이 궁금증은 제가 그냥 쓰면서 문득 javadoc에서 구현부가 궁금할 수있지 않나 라는 생각에서 비롯된 것 같습니다. 그런데, 어차피 구현은 코드를 보면되고, 만약 코드에 따로 설명이 필요할정도로 복잡하다면, 그건  리팩토링이 필요한 코드이겠네요~!

Reply:
그렇습니다. 깨달음을 얻으셨군요!